### ISSUE TITLE
MD013: Add support for applying reflow fix to footnote definitions

### ISSUE THREAD

**Issue by sisp** [rvben/rumdl#499]
### Problem or motivation

I've noticed that rumdl doesn't reformat overlong footnote definitions with `rumdl check --fix` while, e.g., overlong paragraphs are reformatted.

### Proposed solution

It would be great to have a reflow fix for footnote definitions, too. 🙂

### Alternatives considered

n/a

### Feature category

Auto-fix / formatting

### Example markdown

```markdown
# Test

Lorem ipsum dolor sit amet[^1].

[^1]: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

### Additional context

Consider the above Markdown document and the following `.rumdl.toml` config file:

```toml
[global]
line-length = 80

[MD013]
reflow = true
```

The overlong footnote definition is detected by the linter, but there is no automatic fix available in v0.1.42:

```console
$ rumdl check --fix test.md
test.md:5:81: [MD013] Line length 129 exceeds 80 characters

Issues: Found 1 issues in 1 file (14ms)
```

---

I'd be willing to submit a PR, but I'm afraid I have no experience with Rust. 🙁

**Comment by rvben** [comment]
Fixed in v0.1.50. Footnote definitions are now reflowed, including multi-paragraph footnotes. Code blocks, blockquotes, tables, lists, and link reference definitions inside footnotes are preserved verbatim. Thanks for the suggestion, @sisp!

**Comment by sisp** [comment]
Fantastic, @rvben! Thank you very much! :bow:
